lib/hooks.dart

```diff
@@ -158,6 +158,147 @@ void useEffect(dynamic Function() sideEffect, [List<Object> dependencies]) {
   }
 }
 
+/// The return value of [useReducer].
+///
+/// The current state is available via [state] and action dispatcher is available via [dispatch].
+///
+/// > __Note:__ there are two [rules for using Hooks](https://reactjs.org/docs/hooks-rules.html):
+/// >
+/// > * Only call Hooks at the top level.
+/// > * Only call Hooks from inside a [DartFunctionComponent].
+///
+/// Learn more: <https://reactjs.org/docs/hooks-reference.html#usereducer>.
+class ReducerHook<TState, TActions, TInit> {
+  /// The first item of the pair returned by [React.userReducer].
+  TState _state;
+
+  /// The second item in the pair returned by [React.userReducer].
+  void Function(TActions) _dispatch;
+
+  ReducerHook(TState Function(TState state, TActions action) reducer, TState initialState) {
+    final result = React.useReducer(allowInterop(reducer), initialState);
+    _state = result[0];
+    _dispatch = result[1];
+  }
+
+  /// Constructor for [useReducerLazy], calls lazy version of [React.useReducer] to
+  /// initialize [_state] to the return value of [init(initialArg)].
+  ///
+  /// See: <https://reactjs.org/docs/hooks-reference.html#lazy-initialization>.
+  ReducerHook.lazy(
+      TState Function(TState state, TActions action) reducer, TInit initialArg, TState Function(TInit) init) {
+    final result = React.useReducer(allowInterop(reducer), initialArg, allowInterop(init));
+    _state = result[0];
+    _dispatch = result[1];
+  }
+
+  /// The current state map of the component.
+  ///
+  /// See: <https://reactjs.org/docs/hooks-reference.html#usereducer>.
+  TState get state => _state;
+
+  /// Dispatches [action] and triggers stage changes.
+  ///
+  /// > __Note:__ The dispatch function identity is stable and will not change on re-renders.
+  ///
+  /// See: <https://reactjs.org/docs/hooks-reference.html#usereducer>.
+  void dispatch(TActions action) => _dispatch(action);
+}
+
+/// Initializes state of a [DartFunctionComponent] to [initialState] and creates [dispatch] method.
+///
+/// __Example__:
+///
+/// ```
+/// Map reducer(Map state, Map action) {
+///   switch (action['type']) {
+///     case 'increment':
+///       return {...state, 'count': state['count'] + 1};
+///     case 'decrement':
+///       return {...state, 'count': state['count'] - 1};
+///     default:
+///       return state;
+///   }
+/// }
+///
+/// UseReducerTestComponent(Map props) {
+///   final state = useReducer(reducer, {'count': 0});
+///
+///   return react.Fragment({}, [
+///     state.state['count'],
+///     react.button({
+///       'onClick': (_) => state.dispatch({'type': 'increment'})
+///     }, [
+///       '+'
+///     ]),
+///     react.button({
+///       'onClick': (_) => state.dispatch({'type': 'decrement'})
+///     }, [
+///       '-'
+///     ]),
+///   ]);
+/// }
+/// ```
+///
+/// Learn more: <https://reactjs.org/docs/hooks-reference.html#usereducer>.
+ReducerHook<TState, TActions, TInit> useReducer<TState, TActions, TInit>(
```

Comment:
#nit "Action" should be singular here, so as not to confuse it with flux
```suggestion
ReducerHook<TState, TAction, TInit> useReducer<TState, TAction, TInit>(
```

Same on the next line and in the below function